DiagnosticsExtension/Models/ConnectionStringValidator/EventHubsValidator.cs

```diff
@@ -51,53 +53,110 @@ async public Task<ConnectionStringValidationResult> ValidateAsync(string connect
-                    response.Status = ConnectionStringValidationResult.ResultStatus.AuthFailure;
+                    throw new MalformedConnectionStringException(e.Message, e);
                 }
-                else if (e.Message.Contains("Ip has been prevented to connect to the endpoint"))
+            }
+            else
+            {
+                try
                 {
-                    response.Status = ConnectionStringValidationResult.ResultStatus.Forbidden;
+                    string serviceUriString = Environment.GetEnvironmentVariable(appSettingName + ConnectionStringResponseUtility.FullyQualifiedNamespace);
+                    appSettingClientIdValue = Environment.GetEnvironmentVariable(appSettingName + ConnectionStringResponseUtility.ClientId);
+                    appSettingClientCredValue = Environment.GetEnvironmentVariable(appSettingName + ConnectionStringResponseUtility.Credential);
+                    // Creating client using User assigned managed identity
+                    if (!string.IsNullOrEmpty(appSettingClientIdValue))
+                    {
+                        if (appSettingClientCredValue != ConnectionStringResponseUtility.ValidCredentialValue)
+                        {
+                            throw new ManagedIdentityException(ConnectionStringResponseUtility.ManagedIdentityCredentialMissing);
+                        }
+                        else
+                        {
+                            identityType = ConnectionStringValidationResult.ManagedIdentityType.User;
+                            client = new EventHubProducerClient(serviceUriString, eventHubName, new ManagedIdentityCredential(appSettingClientIdValue));
+                        }
+                    }
+                    // Creating client using System assigned managed identity
+                    else
+                    {
+                        identityType = ConnectionStringValidationResult.ManagedIdentityType.System;
+                        client = new EventHubProducerClient(serviceUriString, eventHubName, new ManagedIdentityCredential());
+                    }
                 }
-                else
+                catch (Exception e)
                 {
-                    response.Status = ConnectionStringValidationResult.ResultStatus.UnknownError;
+                    throw new ManagedIdentityException(e.Message, e);
                 }
-                response.Exception = e;
-            }
 
-            return response;
-        }
+            }
+            await client.GetPartitionIdsAsync();
+            await client.CloseAsync();
 
-        protected async Task<TestConnectionData> TestConnectionStringAsync(string connectionString, string name, string clientId)
-        {
             TestConnectionData data = new TestConnectionData
             {
-                ConnectionString = connectionString,
-                Name = name,
+                ConnectionString = client.ToString(),
```

Comment:
Is TestConnectionData.ConnectionString actually getting used anywhere in the latest code? It might have been used in the previous logic.  Let's remove it if not used anywhere while examining the return value.

Reply:
This TestConnectionData object is allowing us to determine whether the test ran successfully or not. 
If the result of this is empty, then it means an unknown error happened, which is what is being taken back to the angular code.